pkg/dependency/parser/python/uv/parse_testcase.go

```diff
@@ -0,0 +1,116 @@
+package uv
+
+import ftypes "github.com/aquasecurity/trivy/pkg/fanal/types"
+
+var (
+	// docker run --name uv --rm -it python@sha256:e1141f10176d74d1a0e87a7c0a0a5a98dd98ec5ac12ce867768f40c6feae2fd9 sh
+	// wget -qO- https://github.com/astral-sh/uv/releases/download/0.5.8/uv-installer.sh | sh
+	// source $HOME/.local/bin/env
+	// uv init normal && cd normal
+	// uv add requests==2.32.0
+	// apk add jq
+	// uv pip list --format json |jq -c 'sort_by(.name) | .[] | {"ID": (.name + "@" + .version), "Name": .name, "Version": .version}' | sed 's/$/,/' | sed 's/\"\([^"]*\)\":/\1:/g'
+
+	// add a root project
+	// fill in the relationships between the packages
+	uvNormal = []ftypes.Package{
+		{ID: "normal@0.1.0", Name: "normal", Version: "0.1.0", Relationship: ftypes.RelationshipRoot},
+		{ID: "requests@2.32.0", Name: "requests", Version: "2.32.0", Relationship: ftypes.RelationshipDirect},
+		{ID: "certifi@2024.8.30", Name: "certifi", Version: "2024.8.30", Relationship: ftypes.RelationshipIndirect, Indirect: true},
+		{ID: "charset-normalizer@3.4.0", Name: "charset-normalizer", Version: "3.4.0", Relationship: ftypes.RelationshipIndirect, Indirect: true},
+		{ID: "idna@3.10", Name: "idna", Version: "3.10", Relationship: ftypes.RelationshipIndirect, Indirect: true},
+		{ID: "urllib3@2.2.3", Name: "urllib3", Version: "2.2.3", Relationship: ftypes.RelationshipIndirect, Indirect: true},
+	}
+
+	// add a root project
+	uvNormalDeps = []ftypes.Dependency{
+		{ID: "normal@0.1.0", DependsOn: []string{"requests@2.32.0"}},
+		{ID: "requests@2.32.0", DependsOn: []string{"certifi@2024.8.30", "charset-normalizer@3.4.0", "idna@3.10", "urllib3@2.2.3"}},
+	}
+
+	// docker run --name uv --rm -it python@sha256:e1141f10176d74d1a0e87a7c0a0a5a98dd98ec5ac12ce867768f40c6feae2fd9 sh
+	// wget -qO- https://github.com/astral-sh/uv/releases/download/0.5.8/uv-installer.sh | sh
+	// source $HOME/.local/bin/env
+	// uv init large && cd large
+	// uv add asyncio==3.4.3 aiohttp==3.11.10 boto3==1.35.79 fastapi==0.115.6 simplejson==3.19.3 SQLAlchemy==2.0.36 pydantic==2.10.3 uvicorn==0.32.1
+	// uv add --group dev pytest==8.3.4 ruff==0.8.2 coverage==7.6.9 codespell==2.3.0
+	// uv add --group docs mkdocs==1.6.1 pymdown-extensions==10.12
+	// apk add jq
+	// uv pip list --format json |jq -c 'sort_by(.name) | .[] | {"ID": (.name + "@" + .version), "Name": .name, "Version": .version}' | sed 's/$/,/' | sed 's/\"\([^"]*\)\":/\1:/g'
+
+	// add a root project
+	// fill in the relationships between the packages
+	uvLarge = []ftypes.Package{
```

Comment:
We used to add this kind of test when the `Package` struct was very small. Today, it has `Relationship`, `License` and many other fields, which makes it hard to maintain large test cases. Instead, we manually create small test cases to cover edge cases these days. I think we can delete it.

Reply:
Removed https://github.com/aquasecurity/trivy/pull/8080/commits/cc079a7ab4cd9b382c2fd0dee015d048d81ba8f3